pkg/trace/writer/trace_nix_test.go

```diff
@@ -51,141 +52,117 @@ func TestPeriodicFlush(t *testing.T) {
+
 	// Close and stop
 	close(traceChannel)
 	traceWriter.Stop()
 
-	// Then we expect some counts to have been sent to the stats client for each update tick (there should have been
-	// at least 3 ticks)
-	countSummaries := statsClient.GetCountSummaries()
-
-	// Payload counts
-	payloadSummary := countSummaries["datadog.trace_agent.trace_writer.payloads"]
-	assert.True(len(payloadSummary.Calls) >= 3, "There should have been multiple payload count calls")
-	assert.Equal(expectedNumPayloads, payloadSummary.Sum)
-
-	// Traces counts
-	tracesSummary := countSummaries["datadog.trace_agent.trace_writer.traces"]
-	assert.True(len(tracesSummary.Calls) >= 3, "There should have been multiple traces count calls")
-	assert.Equal(expectedNumTraces, tracesSummary.Sum)
-
-	// Spans counts
-	spansSummary := countSummaries["datadog.trace_agent.trace_writer.spans"]
-	assert.True(len(spansSummary.Calls) >= 3, "There should have been multiple spans count calls")
-	assert.Equal(expectedNumSpans, spansSummary.Sum)
-
-	// Bytes counts
-	bytesSummary := countSummaries["datadog.trace_agent.trace_writer.bytes"]
-	assert.True(len(bytesSummary.Calls) >= 3, "There should have been multiple bytes count calls")
-	// FIXME: Is GZIP non-deterministic? Why won't equal work here?
-	assert.True(math.Abs(float64(expectedNumBytes-bytesSummary.Sum)) < 100., "Bytes should be within expectations")
-
-	// Retry counts
-	retriesSummary := countSummaries["datadog.trace_agent.trace_writer.retries"]
-	assert.True(len(retriesSummary.Calls) >= 2, "There should have been multiple retries count calls")
-	assert.True(retriesSummary.Sum >= expectedMinNumRetries)
-
-	// Error counts
-	errorsSummary := countSummaries["datadog.trace_agent.trace_writer.errors"]
-	assert.True(len(errorsSummary.Calls) >= 3, "There should have been multiple errors count calls")
-	assert.Equal(expectedNumErrors, errorsSummary.Sum)
-
-	// Single trace max spans
-	singleMaxSpansSummary := countSummaries["datadog.trace_agent.trace_writer.single_max_spans"]
-	assert.True(len(singleMaxSpansSummary.Calls) >= 3, "There should have been multiple single max spans count calls")
-	assert.Equal(expectedNumSingleMaxSpans, singleMaxSpansSummary.Sum)
+	expectStats := summaryCompareFunc(assert, statsClient)
+	for key, sum := range want {
+		expectStats("datadog.trace_agent.trace_writer."+key, 6, sum, 1)
+	}
+}
+
+func summaryCompareFunc(assert *assert.Assertions, statsClient *testutil.TestStatsClient) func(name string, calls int, sum int, delta float64) {
+	summaries := statsClient.GetCountSummaries()
+	return func(name string, minCalls int, sum int, delta float64) {
+		summary, ok := summaries[name]
+		assert.True(ok, fmt.Sprintf("%s not found", name))
+		calls := len(summary.Calls)
+		assert.True(calls >= minCalls, fmt.Sprintf("%s: expected at least %d calls, got %d", name, minCalls, calls))
+		assert.InDelta(sum, summary.Sum, delta, fmt.Sprintf("%s: expected %d, got %d", name, sum, summary.Sum))
```

Comment:
Note to reviewers: I guess this should maybe not be using delta at all and be exact. I've noticed the "retries" are flaky. I'll try and figure it out tomorrow and turn this into an "equal"